libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -373,7 +370,19 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 		make_pair("scope", idOrNull(_node.scope()))
 	};
 
-	if (_node.isPartOfExternalInterface())
+
+	if (_node.annotation().contract)
+	{
+		Visibility visibility;
+		if (_node.isConstructor())
+			visibility = _node.annotation().contract->abstract() ? Visibility::Internal : Visibility::Public;
+		else
+			visibility = _node.visibility();
```

Comment:
If it's not a constructor - do we need the contract to be available?

Reply:
Fair point. Nope, no contract needed in that case. Will adjust.